microsoftgraph/msgraph-cli-archived
# Add extension command module

## Overview

Adds a command to manage 3rd party extensions in the cli

### Demo

```
(msgraph-cli) > ./mg extension list
[]

(msgraph-cli) >./mg extension show
ERROR: the following arguments are required: --name/-n

TRY THIS:
az extension show --name MyExtension
Show an extension. (autogenerated)

az extension list
List the installed extensions. (autogenerated)

https://docs.microsoft.com/en-US/cli/azure/extension#az_extension_show
Read more about the command in reference docs

(msgraph-cli) >./mg extension list-available
[
  {
    "experimental": false,
    "installed": false,
    "name": "mail_beta",
    "preview": true,
    "summary": "Microsoft Azure Command-Line Tools Mail Extension",
    "version": "0.1.0"
  }
]

```

## Testing Instructions

* Run the command `./mg extension --help` to see the available commands
* Run any of the commands to test.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/msgraph-cli/pull/153)
